flink-runtime/src/main/java/org/apache/flink/runtime/rpc/akka/AkkaRpcActor.java

```diff
@@ -263,14 +274,41 @@ private void handleRpcInvocation(RpcInvocation rpcInvocation) {
 								if (throwable != null) {
 									promise.failure(throwable);
 								} else {
-									promise.success(value);
+									long resultSize = 0;
+									if (isRemoteSender) {
+										try {
+											resultSize = new SerializedValue(value).getByteArray().length;
+											if (resultSize > maximumFramesize) {
+												promise.failure(new Throwable("The result size exceeds the maximum " +
+													"size, result size is " + resultSize + ", maximum frame size is " +
+													maximumFramesize + " ."));
+											} else {
+												promise.success(value);
```

Comment:
I think it would be better to send the `SerializedValue`. That way we save some extra serialization work. This, of course, requires that the receiver expects a `SerializedValue`.